ansible/files/archives/opt-custom/method/desktop-init

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+
+# source config vars
+. /opt/custom/etc/desktop.conf
+
+# replace first ^ (match beginning) to space because the vmadm ALIAS column is not the first one
+DESKTOP_NAME_REGEX="$(echo ${DESKTOP_NAME_REGEX} | sed -e 's/^\^/ /')"
+
+
+##### FUNCTIONS #####
+
+create_xuser() {
+  # Recreate X user if necessary (his home in /opt/xuser is persistent)
+  mkdir -p "/opt/${XUSER}"
+  
+  if ! /usr/bin/grep -q "^${XUSER}:" /etc/passwd; then
+    echo "${XUSER}:x:${XUSERID}:1::/opt/${XUSER}:/usr/bin/bash"   >> /etc/passwd
+  fi
+  
+  # Verify shadow separately because it can be loop-mounted /usbkey/shadow
+  if ! /usr/bin/grep -q "^${XUSER}:" /etc/shadow; then
+    echo "${XUSER}:*LK*:::::::"                                   >> /etc/shadow
+  fi
+  
+  chown -R "${XUSER}" "/opt/${XUSER}"
+}
+
+update_VM_IP() {
+  UUID="$(/usr/sbin/vmadm list | /usr/bin/awk "/running.*${DESKTOP_NAME_REGEX}/ {print \$1}" | head -1)"
```

Comment:
Please use vmadm's parsable output without the header and also please consider used a lookup parameter.